File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Unit/Monster/Model/TheBottom/AcidSlime_L_Model.ts
import { Mathf } from "../../../../../../../Core/Module/Math/Mathf";
import DungeonManager from "../../../../DungeonManager";
import FileHelper from "../../../../FileHelper";
import { ILoader } from "../../../../Loader/ILoader";
import { SpineLoader } from "../../../../Loader/SpineLoader";
import { AbstractPower } from "../../../../Power/AbstractPower";
import { SplitPower } from "../../../../Power/SplitPower";
import { MonsterInfo } from "../../MonsterInfo";

export class AcidSlime_L_Model extends MonsterInfo{
    public static Id = "AcidSlime_L"
    private loader : ILoader;
    get InitPower(): Array<AbstractPower> {
        return new Array<AbstractPower>(new SplitPower())
    }
    get MaxHealth(): number {
        if(DungeonManager.Inst.AdvanceLevel >= 7)
            return Mathf.RandomRange(68, 72);
        else
            return Mathf.RandomRange(65, 69);
    }
    get Loader(): ILoader {
        return this.loader ?? (this.loader = new SpineLoader(FileHelper.FormatPath("Monsters/theBottom/slimeL/skeleton.atlas"),FileHelper.FormatPath("Monsters/theBottom/slimeL/skeleton.json"),FileHelper.FormatPath("Monsters/theBottom/slimeL/skeleton.png")));
    }
    get Name(): string {
        return "酸液史莱姆（大）";
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Unit/Monster/Model/TheBottom/AcidSlime_L_Model.cs
using System.Collections.Generic;
using ModTheSpire.Dungeon;
using ModTheSpire.Power;
using ModTheSpire.Unit.Loader;
using UnityEngine;

namespace ModTheSpire.Unit.Monster.Model.TheBottom
{
    public class AcidSlime_L_Model : MonsterInfo
    {
        public static string Id = "AcidSlime_L";
        private ILoader loader;
        //List<AbstractPower> InitPower => new List<AbstractPower>(new SplitPower());

        public override int MaxHealth => DungeonManager.Inst.AdvanceLevel >= 7 ? Random.Range(68, 72) : Random.Range(65, 69);

        public override ILoader Loader
        {
            get
            {
                return this.loader ??= new SpineLoader("Monsters/theBottom/slimeL/skeleton.atlas",
                    "Monsters/theBottom/slimeL/skeleton.json", "Monsters/theBottom/slimeL/skeleton.png");
            }
        }

        public override string Name => "酸液史莱姆（大）";

        public AcidSlime_L_Model(Vector2 pos) : base(pos)
        {
        }
    }
}